Core/Prognetics.Saga.Parsers.DependencyInjection/ParserExtensions.cs

```diff
@@ -3,37 +3,36 @@
 using Prognetics.Saga.Core.Abstract;
 using Prognetics.Saga.Core.DependencyInjection;
 using Prognetics.Saga.Parsers.Core.Model;
-using System.Reflection;
 
 namespace Prognetics.Saga.Parsers.DependencyInjection
 {
     public static class ParserExtensions
     {
-        public static ISagaConfiguration UseParser(this ISagaConfiguration configuration,
+        public static ISagaConfiguration UseParser(
+            this ISagaConfiguration configuration,
             Action<ModelSourceOptions> configureOptions)
         {
             configuration.Services.Configure(configureOptions);
             configuration.Services.AddTransient(x => x.GetRequiredService<IOptions<ModelSourceOptions>>().Value);
                          
-            configuration.Services.AddSingleton<IEnumerable<IModelSource>>(provider =>
-            {
-                var result = new List<IModelSource>();
-                var options = provider.GetRequiredService<ModelSourceOptions>();
-
-                foreach (var confirguration in options.Configurations)
-                {
-                    var type = Type.GetType(confirguration.ParserType, true);
```

Comment:
I understand that you have replaced the reflection with the registration of a specific parser. Can you please elaborate on how this will work from the user's perspective now?

Reply:
Also can such changes in the future can be part of a separate PR? Just for easier review :)

Reply:
Its the same code but refactored to use aggregate instead of foreach. There are less curly brackets so i in my opinion is more readable. In the other hand is just a little bit slower so If you dont like using LINQ in this way i can rollback this change, otherwise I`ll move the code to the PR seperate PR.